res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,718 @@
+// TODO(sblaisot): Define how to change mode from controller
+var FxMode = 1; // Single effect mode by default
+
+var ON = 0x7F,
+    OFF = 0x00,
+    RED = 0x7F,
+    BLUE = 0x55,
+    VIOLET = 0x2A,
+    SHIFT = 0x40,
+    DOWN = 0x7F,
+    UP = 0x00;
+
+// Initialise and shutdown stuff.
+// ========================================================
+ReloopBeatmix24.TurnLEDsOff = function() {
+    // Turn all LEDS off
+    var i, j;
+    for (i = 0x91; i <= 0x94; i++) { // 4 decks
+
+        // Pads
+        for (j = 0x00; j <= 0x0F; j++) {
+            midi.sendShortMsg(i, j, OFF);
+            midi.sendShortMsg(i, j + SHIFT, OFF);
+        }
+        for (j = 0x10; j <= 0x1F; j++) {
+            midi.sendShortMsg(i, j, OFF);
+        }
+
+        // Play/cue/cup/sync/pitch bend
+        for (j = 0x20; j <= 0x25; j++) {
+            midi.sendShortMsg(i, j, OFF);
+            midi.sendShortMsg(i, j + SHIFT, OFF);
+        }
+
+        // Jog Wheel leds
+        ReloopBeatmix24.AllJogLEDsToggle(i, OFF);
+
+        // Load + Cue
+        midi.sendShortMsg(i, 0x50, OFF);
+        midi.sendShortMsg(i, 0x52, OFF);
+        midi.sendShortMsg(i, 0x72, OFF);
+    }
+};
+
+ReloopBeatmix24.connectControls = function(disconnect) {
+
+    // Channels controls
+    for (var i = 1; i <= 4; i++) {
+        engine.connectControl("[Channel" + i + "]", "track_samples",
+            "ReloopBeatmix24.deckloaded", disconnect);
+        engine.connectControl("[Channel" + i + "]", "play",
+            "ReloopBeatmix24.startJogLedSpinnie", disconnect);
+        engine.connectControl("[Channel" + i + "]", "loop_end_position",
+            "ReloopBeatmix24.loopDefined", disconnect);
+        engine.softTakeover("[Channel" + i + "]", "rate", disconnect ?
+            false : true);
+        engine.setValue("[EffectRack1_EffectUnit1]", "group_[Channel" + i +
+            "]_enable", 0);
+        engine.setValue("[EffectRack1_EffectUnit2]", "group_[Channel" + i +
+            "]_enable", 0);
+    }
+
+    // Samplers controls
+    for (i = 1; i <= 8; i++) {
+        engine.connectControl("[Sampler" + i + "]", "track_samples",
+            "ReloopBeatmix24.deckloaded", disconnect);
+        engine.connectControl("[Sampler" + i + "]", "play",
+            "ReloopBeatmix24.SamplerPlay", disconnect);
+    }
+
+    // Effects reset
+    engine.setValue("[EffectRack1_EffectUnit1]", "group_[Master]_enable", 0);
+    engine.setValue("[EffectRack1_EffectUnit2]", "group_[Master]_enable", 0);
+};
+
+ReloopBeatmix24.init = function(id, debug) {
+    ReloopBeatmix24.id = id;
+    ReloopBeatmix24.TurnLEDsOff(); // Turn off all LEDs
+    ReloopBeatmix24.connectControls(false);
+
+    for (var i = 1; i <= 4; i++) {
+        engine.trigger("[Channel" + i + "]", "loop_end_position");
+    }
+
+    print("Reloop Beatmix: " + id + " initialized.");
+};
+
+ReloopBeatmix24.shutdown = function() {
+    ReloopBeatmix24.TurnLEDsOff(); // Turn off all LEDs
+    ReloopBeatmix24.connectControls(true);
```

Comment:
I'm sure you're right at shutdown, but is this also true when changing the mapping of a controller ?